src/metaopt/worker/experiment.py

```diff
@@ -0,0 +1,354 @@
+            user=self.metadata['user'],
+            status='completed'
+            )
+        num_completed_trials = len(self._db.read('trials', query, {'_id': 1}))
+        if num_completed_trials >= self.max_trials:
+            return True
+        return False
+
+    @property
+    def configuration(self):
+        """Return a copy of an `Experiment` configuration as a dictionary."""
+        if self._init_done:
+            return self._db.read('experiments', {'_id': self._id})[0]
+
+        config = dict()
+        for attrname in self.__slots__:
+            if not attrname.startswith('_'):
+                config[attrname] = getattr(self, attrname)
+        # Reason for deepcopy is that some attributes are dictionaries
+        # themselves, we don't want to accidentally change the state of this
+        # object from a getter.
+        return copy.deepcopy(config)
+
+    def configure(self, config):
+        """Set `Experiment` by overwriting current attributes.
+
+        If `Experiment` was already set and an overwrite is needed, a *fork*
+        is advised with a different :attr:`name` for this particular configuration.
+
+        .. note:: Calling this property is necessary for an experiment's
+           initialization process to be considered as done. But it can be called
+           only once.
+        """
+        if self._init_done:
+            raise RuntimeError("Configuration is done; cannot reset an Experiment.")
+
+        # If status is None in this object, then database did not hit a config
+        # with same (name, user's name) pair. Everything depends on the user's
+        # moptconfig to set.
+        if self.status is None:
+            if config['name'] != self.name or \
+                    config['metadata']['user'] != self.metadata['user'] or \
+                    config['metadata']['datetime'] != self.metadata['datetime']:
+                raise ValueError("Configuration given is inconsistent with this Experiment.")
+            is_new = True
+        else:
+            # Fork if it is needed
+            is_new = self._is_different_from(config)
+            if is_new:
+                self._fork_config(config)  # Change (?) `name` attribute here.
+
+        # Just overwrite everything given
+        for section, value in six.iteritems(config):
+            if section == 'status' or \
+                    section not in self.__slots__ or \
+                    section.startswith('_'):
+                continue
+            setattr(self, section, value)
+
+        self.status = 'new'
+        final_config = self.configuration  # grab dict representation of Experiment
+
+        # Sanitize and replace some sections with objects
+        self._sanitize_config()
+
+        # If everything is alright, push new config to database
+        if is_new:
+            self._db.write('experiments', final_config)
+            # XXX: This may be MongoDB only; it updates an inserted dict with _id
```

Comment:
You are right. I will keep this as a reminder though, for later, while appending your comment on this. Tests are supposed to work like this. So we do not need to write more tests, if were to copy them to a new implementation. (I say copy because of this #23)